cloudsync/api_test.py

```diff
@@ -130,3 +150,148 @@ def test_process_transcode_results(mocker):
     process_transcode_results(video, job)
     assert len(video.videofile_set.all()) == 2
     assert len(video.videothumbnail_set.all()) == 1
+
+
+def test_extract_title():
+    """
+    Test that a correctly formatted title is returned for a video file.
+    """
+    filenames = (('MIT-6.046-2017-Spring-lec-mit-0000-2017apr06-0404-L01.mp4', 'MIT-6.046-2017-Spring-L01'),
+                 ('MIT-6.046-2018-Fall-lec-mit-0000-2017apr06-0404.mp4', 'MIT-6.046-2018-Fall'),
+                 ('MIT-6.046-lec-mit-0000-2017apr06-0404.mp4', 'MIT-6.046'),
+                 ('MIT-6.046-lec-mit-0000-2017apr06-0404-L02.mp4', 'MIT-6.046-L02'),
+                 ('MIT-STS.001-2015-Fall-lec-mit-0000-2015dec21-2222-L03_2.mp4', 'MIT-STS.001-2015-Fall-L03'),
+                 ('MIT-HST.969-2018-Spring-lec-mit-0000-2018mar09-0001-L01.mp4', 'MIT-HST.969-2018-Spring-L01'),
+                 ('NUS-7.S939-2019-Spring-lec-mit-0000-2019apr02-1210-L05.mp4', 'NUS-7.S939-2019-Spring-L05'),
+                 ('MIT-Adhoc-lec-mit-0000-2017aug10-0954-2-190.mp4', 'MIT-Adhoc-2-190'),
+                 ('MIT-18.test-lec-mit-0000-2017aug10-0956-L01.mp4', 'MIT-18.test-L01'),
+                 ('MIT-18.test-lec-mit-0000-2017aug10-0956.mp4', 'MIT-18.test'),
+                 ('MIT-Simons Lecture Series-lec-mit-0000-2017aug10-0956-2-10.mp4', 'MIT-Simons Lecture Series-2-10'),
+                 ('/&*3:<>俺正和-lec-mit-0000-2017aug10-0956.mp4', '/&*3:<>俺正和'))
+    for item in filenames:
```

Comment:
nit, please unpack the tuple with meaningful variable names, so something like
`for input, output in filenames:`